python_modules/dagster/dagster/_core/definitions/unresolved_asset_sensor_definition.py

```diff
@@ -0,0 +1,279 @@
+        asset can have the most up to date data. So, for each parent asset we check if it has a planned asset
+        materialization event that is newer than the latest complete materialization. If this is the case, we
+        don't want the asset to materialize. So we set parent_asset_event_records to False for all
+        parents (so that if the sensor is set to materialize if any of the parents are updated, the sensor will
+        still choose to not materialize the asset) and immediately return.
+        """
+        from dagster._core.events import DagsterEventType
+        from dagster._core.storage.event_log.base import EventRecordsFilter
+
+        parent_asset_event_records = {}
+
+        for p in parent_assets:
+            print(f"Materializations for parent {p}")
+            if p in will_materialize_list:
+                # if the parent will be materialized by this sensor, then we can also
+                # materialize the child
+                parent_asset_event_records[p] = (True, None)
+            elif p in self._source_asset_keys:
+                # if the parent is a source asset, we assume it has always been updated
+                # this will be replaced if we introduce a versioning schema for source assets
+                print("parent is a source asset")
+                parent_asset_event_records[p] = (True, None)
+            else:
+                # if the parent has been materialized and is not currently being materialized, then
+                # we want to materialize the child
+
+                # get the most recent completed materialization
+                event_records = context.instance.get_event_records(
+                    EventRecordsFilter(
+                        event_type=DagsterEventType.ASSET_MATERIALIZATION,
+                        asset_key=p,
+                        after_cursor=cursor.get(str(p)),
+                    ),
+                    ascending=False,
+                    limit=1,
+                )
+                print(f"materialization record {event_records}")
+
+                # get the most recent planned materialization
+                materialization_planned_event_records = context.instance.get_event_records(
+                    EventRecordsFilter(
+                        event_type=DagsterEventType.ASSET_MATERIALIZATION_PLANNED,
+                        asset_key=p,
+                    ),
+                    ascending=False,
+                    limit=1,
+                )
+
+                print(f"will materialize record {materialization_planned_event_records}")
+
+                if event_records and materialization_planned_event_records:
+                    # planned materialization is from a different run than the most recent
+                    # completed materialization
+                    if (
+                        event_records[0].event_log_entry.run_id
```

Comment:
I think it's possible that a run will yield an `ASSET_MATERIALIZATION_PLANNED` event and then fail before the asset is actually materialized. I think we should instead test if the run of the `ASSET_MATERIALIZATION_PLANNED` event is in progress instead of testing equivalence by run id:
```
instance.get_runs(filters=RunsFilter(run_ids=[...], statuses=IN_PROGRESS_RUN_STATUSES))
```

Reply:
oh cool that's way better, didn't think/know to do that

Reply:
this fixed the issue in the test suite too. thanks for the suggestion!